localstack/services/providers.py

```diff
@@ -16,13 +16,14 @@ def acm():
 
 @aws_provider()
 def apigateway():
-    from localstack.services.apigateway import apigateway_listener, apigateway_starter
+    from localstack.services.apigateway.patches import apply_patches
+    from localstack.services.apigateway.provider import ApigatewayApiListener, ApigatewayProvider
 
-    return Service(
-        "apigateway",
-        listener=apigateway_listener.UPDATE_APIGATEWAY,
-        start=apigateway_starter.start_apigateway,
-    )
+    provider = ApigatewayProvider()
+    apply_patches()
```

Comment:
nit; AFAIK we can use the lifecycle hooks to apply the patches? something we can iterate anyway

Reply:
Agreed - cleaner to have it in the hooks. Updated 👍